src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java

```diff
@@ -4,131 +4,260 @@
 import com.minecolonies.colony.CitizenData;
 import com.minecolonies.colony.Colony;
 import com.minecolonies.colony.ColonyView;
+import com.minecolonies.colony.Field;
 import com.minecolonies.colony.jobs.AbstractJob;
 import com.minecolonies.colony.jobs.JobFarmer;
-import io.netty.buffer.ByteBuf;
+import com.minecolonies.tileentities.ScarecrowTileEntity;
 import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
 import net.minecraft.util.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.Constants;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Class which handles the farmer building.
+ */
 public class BuildingFarmer extends AbstractBuildingWorker
 {
+    /**
+     * Descriptive string of the profession.
+     */
+    private static final    String  FARMER      = "Farmer";
 
-    public                  int     wheat    = 100;
-    public                  int     potato   = 0;
-    public                  int     carrot   = 0;
-    public                  int     melon    = 0;
-    public                  int     pumpkin  = 0;
+    /**
+     * The maximum building level of the hut.
+     */
+    private static final    int     MAX_BUILDING_LEVEL = 3;
 
-    public  static final    String  WHEAT_TAG    = "wheat";
-    public  static final    String  POTATO_TAG   = "potato";
-    public  static final    String  CARROT_TAG   = "carrot";
-    public  static final    String  MELON_TAG    = "melon";
-    public  static final    String  PUMPKIN_TAG  = "pumpkin";
+    /**
+     * NBTTag to store the fields.
+     */
+    private static final String TAG_FIELDS = "fields";
 
-    private static final    String  FARMER      = "Farmer";
-    private static final    String  TAG_FARMER  = "farmer";
+    /**
+     * NBTTag to store the currentField.
+     */
+    private static final String TAG_CURRENT_FIELD = "currentField";
+
+    /**
+     * The list of the fields the farmer manages.
+     */
+    private ArrayList<Field> farmerFields = new ArrayList<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private field 'farmerFields' could be made final; it is only initialized in the declaration or constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AImmutableField)